### ISSUE TITLE
Programmatically add to wallet

### ISSUE THREAD

**Issue by petermorlion** [wpswings/wallet-system-for-woocommerce#194]
I'm looking for a simple wallet plugin, but I need to be able to add to the wallet programmatically. I'm developing a custom plugin for a customer, and depending on certain actions/events in my plugin, I need to add funds to a user's wallet. So in code, instead of allowing the user to do so. Is this possible?

**Comment by WP-Swings** [comment]
Hi @petermorlion ,
Thanks for putting your query here.
We have independent function through which you can credit or debit amount from users wallet. Here are the functions name which you can use: **wps_wsfw_update_user_wallet_balance - debit
wps_wsfw_credit_user_wallet_balance -- credit**

The parameters for these functions are required ($user_id, $amount, $order_id = '').
Hope this will work for you as per your need .
Best Regards

**Comment by petermorlion** [comment]
Thank you for your response!

**Comment by petermorlion** [comment]
I've finally gotten round to implementing my feature which uses these methods. Except, they don't seem to do anything. On closer inspection, I notice two "weird" things:

1. These functions aren't used by your plugin
2. Inside the function, there is a check to see if the `wps_wallet` user meta isn't empty. On first run, this metadata will probably be "empty" because the record hasn't been added to the database yet, right?

So could it be these functions won't ever work?

**Comment by WP-Swings** [comment]
Hello @petermorlion ,

Here are the answers to your queries:

1. We created these functions specifically for client support so that clients can extend the functionality according to their needs.
2. We have fixed this issue in our branch. You can copy the code of that file from here: [class-wallet-system-for-woocommerce-dependency.php](https://github.com/wpswings/wallet-system-for-woocommerce/blob/v2.5.14/includes/class-wallet-system-for-woocommerce-dependency.php).

**Comment by petermorlion** [comment]
Yes, that works! I assume 2.5.14 will be available soon (it's not available now, is that due to the recent security issue)?
